src/client.rs

```diff
@@ -92,7 +92,13 @@ impl<'a, T: Authenticator + 'a> CDRS<T> {
             let body = start_response.get_body();
             let authenticator = body.get_authenticator().unwrap();
 
-            if authenticator.as_str() == self.authenticator.get_cassandra_name() {
+            if self.authenticator.get_cassandra_name() == "NONE" {
```

Comment:
not exactly a subtle changes 
1. 1st if if client don't provide any authenticators but server says other wise 
2.  Client provides an authenticator and `auth scheme` provided by the server matches with what client provided then proceed.
3.   Client provides an authenticator and `auth scheme` provided by the server **DOESN'T** match then return an error _Unsupported type of authenticator. {:?} got, but {} is supported_